Tests/SwiftAlgorithmsTests/LazySplitCollectionTests.swift

```diff
@@ -0,0 +1,226 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Algorithms open source project
+//
+// Copyright (c) 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+import Algorithms
+import XCTest
+
+final class LazySplitCollectionTests: XCTestCase {
```

Comment:
For the collection wrapper it may be a good idea to have a test that executes `validateIndexTraversals` just to ensure correctness :)

Reply:
Ah, thank you. I had seen that in another type's test, but at the time it didn't apply to this. Now it does--thank you for reminding me.

Reply:
I learned `validateIndexTraversals` requires `BidirectionalCollection` conformance, which I haven't implemented yet, so adding this test will have to wait a bit.

Reply:
We can/should break that test into versions that work for regular collections and also bidirectional ones, so no worries on this one. 

ALSO: I don't think there's a way to efficiently implement bidirectionality here, because of `maxSplits`. You'd need to traverse from the beginning every time, defeating the point.

Reply:
Ah, I see. You're saying the collection needs to be the same forwards and backwards, right?

Reply:
I think the issue is that with bidirectional conformance we would have to compute the `splits` form the last to first (implementing index before), but because of `maxSplits` we can't compute that without knowing the initial splits, so it will always have to start from `startIndex` which I agree with @natecook1000, defeats the point of implementing the conformance since we can't conceptually traverse from `last` to `first`... at least to my understanding

Reply:
Exactly! Because `maxSplits` can limit the number of subsequences that are produced, there's no way to calculate the last subsequence without searching from the beginning. `BidirectionalCollection` conformance is for collections that can efficiently do backward traversal, so there's just no need to do that here.